client/async.go

```diff
@@ -165,20 +169,19 @@ func (a *gRPCClientAsync) metricsReporterInterceptor(
 				err.Error(),
 			).Inc()
 		}
-	} else {
-		for _, e := range events {
-			metrics.ClientRequestsSuccessCounter.WithLabelValues(
-				hostname,
-				method,
-				e.Topic,
-			).Inc()
-		}
 	}
-	return err
+	for _, e := range events {
+		metrics.ClientRequestsSuccessCounter.WithLabelValues(
+			hostname,
+			method,
+			e.Topic,
+		).Inc()
+	}
+	return nil
 }
 
 func (a *gRPCClientAsync) send(ctx context.Context, event *pb.Event) error {
-	atomic.AddUint64(&a.transientCounter, 1)
+	a.wg.Add(1)
```

Comment:
Good call with the WaitGroup here, much better.

Reply:
thx :D